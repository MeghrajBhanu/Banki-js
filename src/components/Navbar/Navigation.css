.nav-link:hover{
    color:#f8fc2d !important;
}
.nav-link:active{
    color:yellow !important;
    font-weight: 700 !important;
}
.nav-link {
    display:inline-block !important;
    color:white !important;
    font-size: 1rem !important;
    text-decoration: none !important;
}
.text-dec{
    list-style: none !important;
}
.nav-item{
    margin:0 .625rem !important;
    list-style: none !important;
    
}
.active{
    color:yellow !important;
    font-weight: 700 !important;
    text-shadow: #f8fc2d !important;
}