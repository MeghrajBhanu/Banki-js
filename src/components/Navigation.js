import React from "react";
import { Link, NavLink } from "react-router-dom";
import { useSelector } from "react-redux";
import { logout } from "../redux/actions/auth";
import { useDispatch } from "react-redux";
import { useNavigate } from "react-router";

/**
 * Navigation is a UI component that is used for navigation to pages
 *
 * @returns {React.ReactElement}
 */
const Navigation = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const isLoggedIn = useSelector((store) => store.auth.isLoggedIn);
  const handleClick = (e) => {
    e.preventDefault();
    dispatch(logout());
    alert("userlogged out");
    navigate("/");
  };
  return (
    <>
      <nav className="navbar navbar-expand-md navbar-light bg-dark bg-gradient navbar-static-top  ">
        <div className="container">
          <Link className="navbar-brand nav-link active" to="/">
            <span className="text-light">App</span>
          </Link>
          <button
            className="navbar-toggler"
            type="button"
            data-bs-toggle="collapse"
            data-bs-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon"></span>
          </button>
          <div className="collapse navbar-collapse" id="navbarSupportedContent">
            <ul className="navbar-nav align-items-center ms-auto mb-2 mb-lg-0">
              <li className="nav-item">
                <NavLink className="nav-link active" to="/">
                  <span className="text-light">Home</span>
                </NavLink>
              </li>
              {!isLoggedIn && (
                <li className="nav-item ms-lg-3">
                  <NavLink className="nav-link active" to="/login">
                    <span className="text-light">Login</span>
                  </NavLink>
                </li>
              )}
              {!isLoggedIn && (
                <li className="nav-item">
                  <NavLink className="nav-link active" to="/register">
                    <span className="text-light">Register</span>
                  </NavLink>
                </li>
              )}
              {isLoggedIn && (
                <li className="nav-item">
                  <NavLink className="nav-link active" to="/login/landing">
                    <span className="text-light">Landing</span>
                  </NavLink>
                </li>
              )}
              {isLoggedIn && (
                <li className="nav-item">
                  <NavLink className="nav-link active" to="/login/flagged">
                    <span className="text-light">Flagged</span>
                  </NavLink>
                </li>
              )}
              {isLoggedIn && (
                <li className="nav-item">
                  <button className="btn btn-primary" onClick={handleClick}>
                    Logout
                  </button>
                </li>
              )}
            </ul>
          </div>
        </div>
      </nav>
    </>
  );
};
export default Navigation;
