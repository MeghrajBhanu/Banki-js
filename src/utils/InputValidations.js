export const validateEmail = (checkingText) => {
    
    const regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/; 
   
    
    if (regexp.exec(checkingText) !== null) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Email Address'
            };
        }
}
export const validatePan = (checkingText) => {
    
    
    if (checkingText.length === 10) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Pan Number'
            };
        }
}
export const validatePass = (checkingText) => {
    
    
    if (checkingText!=="") {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Password'
            };
        }
}