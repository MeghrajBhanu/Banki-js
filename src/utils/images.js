import bank1 from "../assets/bank1.jpg";
import bank2 from "../assets/bank2.png";
import bank3 from "../assets/bank3.jfif";
import bank4 from "../assets/bank4.png";

const image_map={
    "Bank1":bank1,
    "Bank2":bank2,
    "Bank3":bank3,
    "Bank4":bank4,
}
export default image_map;