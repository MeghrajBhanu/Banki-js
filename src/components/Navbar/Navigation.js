import React, { useState } from "react";
import { Link, NavLink } from "react-router-dom";
import { logout } from "../../redux/actions/auth";
import { useDispatch,useSelector } from "react-redux";
import { useNavigate } from "react-router";
import  './Navigation.css';
import "react-toastify/dist/ReactToastify.css";
/**
 * Navigation is a UI component that is used for navigation to pages
 *
 * @returns {React.ReactElement}
 */
const Navigation = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const token=useSelector(state=>state.auth.token)
  const [clicked,setClicked]=useState(false);

  const isLoggedIn = useSelector((store) => store.auth.isLoggedIn);
  const handleClick = (e) => {
    //e.preventDefault();
    setClicked(true)
    console.log(token)
    
    dispatch(logout(token));
    
    alert("User Logged out");
    
    navigate("/");
    
  };
  return (
    <>
      
      <nav className="navbar navbar-expand-md navbar-light bg-dark bg-gradient sticky-top">
        <div className="container">
          <Link className="navbar-brand nav-link" to="/">
            App
          </Link>
          <button
            className="navbar-toggler"
            type="button"
            data-bs-toggle="collapse"
            data-bs-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon"></span>
          </button>
          <div className="collapse navbar-collapse" id="navbarSupportedContent">
            <ul className="navbar-nav align-items-center ms-auto mb-2 mb-lg-0">
            {!isLoggedIn && (
              <li className="nav-item">
                <NavLink className='nav-link' to="/">
                  Home
                </NavLink>
              </li>)}
              {!isLoggedIn && (
                <li className="nav-item">
                  <NavLink className='nav-link' to="/login">
                    Login
                  </NavLink>
                </li>
              )}
              {!isLoggedIn && (
                <li className="nav-item">
                  <NavLink className="nav-link" to="/register">
                    Register
                  </NavLink>
                </li>
              )}
              {isLoggedIn && (
                <li className="nav-item">
                  <NavLink className="nav-link" to="/login/landing">
                    Landing
                  </NavLink>
                </li>
              )}
              {isLoggedIn && (
                <li className="nav-item">
                  <NavLink className="nav-link" to="/login/flagged">
                    Flagged
                  </NavLink>
                </li>
              )}
              {isLoggedIn && (
                <li className="nav-item">
                  <NavLink className="nav-link" to="/login/banks">
                    Banks
                  </NavLink>
                </li>
              )}
              {isLoggedIn && (
                <li className="nav-item">
                  <button className="btn btn-primary" onClick={(e)=>{
                    
                    handleClick(e)

                  }}
                    >
                    Logout
                  </button>
                </li>
              )}
            </ul>
          </div>
        </div>
      </nav>
    </>
  );
};
export default Navigation;
