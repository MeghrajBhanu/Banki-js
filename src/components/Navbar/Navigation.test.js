import { render, screen, waitFor } from "@testing-library/react";
import { onlywrap, renderComponent } from "../../utils/test_helper";
import renderer from "react-test-renderer";
import Navigation from "./Navigation";

describe("Navigation", () => {
  test("snapshot test", () => {
    const tree = renderer.create(onlywrap(<Navigation />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});