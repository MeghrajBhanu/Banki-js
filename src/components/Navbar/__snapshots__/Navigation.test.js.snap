// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation snapshot test 1`] = `
<nav
  className="navbar navbar-expand-md navbar-light bg-dark bg-gradient sticky-top"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand nav-link"
      href="/"
      onClick={[Function]}
    >
      App
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarSupportedContent"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav align-items-center ms-auto mb-2 mb-lg-0"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
